doc/sai-ptf/pcbb_test_plan.md

```diff
@@ -0,0 +1,346 @@
+3. Set dest ports with lossless buffer pool profile
+4. According to DSCP_MAP_TABLE check the corrosponding priority_group packets stats on the possible ports(use sai_thrift_get_ingress_priority_group_stats with "SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS")
+5. Get the packet number for different meanings, pkts_num_egr_mem for filling up memory, pkts_num_leak_out for leakout packets, pkts_num_trig_pfc for triggering PFC and margin for a tolerance ( reference [config](https://github.com/Azure/sonic-mgmt/blob/master//ansible/vars/qos.yml) which use in case [PFCTest](https://github.com/Azure/sonic-mgmt/blob/master/tests/saitests/sai_qos_tests.py#665))
+6. Disable ports (set port SAI_PORT_ATTR_PKT_TX_ENABLE)
+7. Send N packets from lag1 to fill up the shared buffer (pkts_num_egr_mem + pkts_num_leak_out + pkts_num_trig_pfc - 1 - margin)
+8. Recording the counters on the receiving and transmit ports
+9. Send the packets again
+10. Verify PFC drop happened(Receive port counters are larger than transmit port counters).
+
+- decap_ecn_no_congestion:
+
+1. Make sure LAGs and NextHop groups set as basic [config](./config_data/config_t0.md) 
+2. Make sure tunnel DSCP in PIPE mode, port and tunnel binding to the DSCP map, queue map and priority map as basic [config](./config_data/config_t0.md)
+3. Make sure set PCBB ECN configurations
+4. Generate 100 packets, take one row from the ``DSCP_MAP_TABLE``, set the ``ip_dscp`` accordingly, according to DSCP_MAP_TABLE set the ``ip_ecn``
+5. Send input ipinip packet from lag1 with ``Inner_ECN`` ``Outer_ECN`` ``Inner DSCP`` and ``Outer dscp`` according to ``DSCP_MAP_TABLE`` (only inner take effect).
+5. Create expected decap packet with ``ECN`` and ``Inner DSCP`` according to ``DSCP_MAP_TABLE`` and ``ECN_TABLE``
+6. Recieve decapped packet from port1. Compare it with the expected decap packet.
+
+- decap_ecn_congestion:
+
+1. Make sure LAGs and NextHop groups set as basic [config](./config_data/config_t0.md) 
+2. Make sure tunnel DSCP in PIPE mode, port and tunnel binding to the DSCP map, queue map, and priority map as basic [config](./config_data/config_t0.md)
+3. Make sure set PCBB ECN configurations
+4. Set all the dest port with lossless buffer pool profile
+5. Disable ports (set port SAI_PORT_ATTR_PKT_TX_ENABLE)
+6. According to DSCP_MAP_TABLE check the corrosponding priority_group packets stats on the possible ports(use sai_thrift_get_ingress_priority_group_stats with "SAI_INGRESS_PRIORITY_GROUP_STAT_PACKETS")
+7. Generate N packets, take one row from the ``DSCP_MAP_TABLE``, set the ``ip_dscp`` accordingly, according to DSCP_MAP_TABLE set the ``ip_ecn``
+8. Use the similar approach in decap_pfc_pause test to fill up the buffer
+8.  Enable ports (set port SAI_PORT_ATTR_PKT_TX_ENABLE)
+9.  Recieve decap packet from port1. Compare it with the expected packet for the ECN and DSCP values.
+
+- decap_global_dscp_to_tc_map:
+
+1. Make sure LAGs and NextHop groups set as basic [config](./config_data/config_t0.md) 
```

Comment:
To verify the functionality of global `DSCP_TO_TC_MAP`, we should run the test on a decap tunnel without the tunnel qos attributes. Otherwise the global `DSCP_TO_TC_MAP` is overwritten.

Reply:
Yes, mentioned in step 2 as
 _config port with ``Port DSCP MAP`` and switch_dscp_to_tc with dscp map in ``Tunnel TC MAP``_

adding more statement for emphasize it